client/src/pages/Welcome.js

```diff
@@ -50,7 +36,22 @@ const HighlightLink = styled.div`
 `;
 
 function Welcome() {
-  const [nameUser, setNameUser] = React.useState('');
+  const history = useHistory();
+  const [nameUser, setNameUser] = React.useState(
+    sessionStorage.getItem('Name') || ''
+  );
+
+  React.useEffect(() => {
+    sessionStorage.setItem('Name', nameUser);
```

Comment:
I'm not sure if i understand the context of this right, but wouldn't it be enough to store the name in the session storage when the user press enter?
Instead of using the useEffect you could make it like this:

```
const keyboardEnter = (event) => {
    if (event.keyCode === 13) {
    sessionStorage.setItem('Name', nameUser);
      history.push('/Genres');
    }
  };
```
In your current version the useEffect gets triggered on every keypress during the user is entering the name.
And why you dont call the variable `userName`?

Reply:
Yes, thank you for your help, you are right. I guess, because I was writing the session storage function way earlier than the keyboard event. The reason why I used nameUser instead of userName is that I want to use userName for signing up and storing userdata in the database in a later issue and hence didn't want to confuse it with just this name.